### ISSUE TITLE
vector-0.11.0.0 fails to build with primitive-0.6.2.0

### ISSUE THREAD

**Issue by RyanGlScott** [haskell/vector#142]
```
$ cabal install vector --constraint=primitive==0.6.2.0
Resolving dependencies...
Configuring vector-0.11.0.0...
Building vector-0.11.0.0...
Failed to install vector-0.11.0.0
Build log ( /home/ryanglscott/.cabal/logs/vector-0.11.0.0.log ):                          
cabal: Entering directory '/tmp/cabal-tmp-26528/vector-0.11.0.0'                          
Configuring vector-0.11.0.0...                                                            
Building vector-0.11.0.0...                                                               
Preprocessing library vector-0.11.0.0...
[ 1 of 21] Compiling Data.Vector.Storable.Internal ( Data/Vector/Storable/Internal.hs, dist/build/Data/Vector/Storable/Internal.o )

Data/Vector/Storable/Internal.hs:17:1: warning: [-Wunused-imports]
    The import of ‘Foreign.ForeignPtr’ is redundant
      except perhaps to import instances from ‘Foreign.ForeignPtr’
    To import instances alone, use: import Foreign.ForeignPtr()

Data/Vector/Storable/Internal.hs:18:1: warning: [-Wunused-imports]
    The import of ‘Foreign.Ptr’ is redundant
      except perhaps to import instances from ‘Foreign.Ptr’
    To import instances alone, use: import Foreign.Ptr()
[ 2 of 21] Compiling Data.Vector.Generic.Mutable.Base ( Data/Vector/Generic/Mutable/Base.hs, dist/build/Data/Vector/Generic/Mutable/Base.o )
[ 3 of 21] Compiling Data.Vector.Generic.Base ( Data/Vector/Generic/Base.hs, dist/build/Data/Vector/Generic/Base.o )
[ 4 of 21] Compiling Data.Vector.Fusion.Util ( Data/Vector/Fusion/Util.hs, dist/build/Data/Vector/Fusion/Util.o )
[ 5 of 21] Compiling Data.Vector.Fusion.Bundle.Size ( Data/Vector/Fusion/Bundle/Size.hs, dist/build/Data/Vector/Fusion/Bundle/Size.o )
[ 6 of 21] Compiling Data.Vector.Internal.Check ( Data/Vector/Internal/Check.hs, dist/build/Data/Vector/Internal/Check.o )
[ 7 of 21] Compiling Data.Vector.Fusion.Stream.Monadic ( Data/Vector/Fusion/Stream/Monadic.hs, dist/build/Data/Vector/Fusion/Stream/Monadic.o )

Data/Vector/Fusion/Stream/Monadic.hs:431:1: warning: [-Wredundant-constraints]
    • Redundant constraints: (Monad m, Monad m')
    • In the type signature for:
           trans :: (Monad m, Monad m') =>
                    (forall z. m z -> m' z) -> Stream m a -> Stream m' a

Data/Vector/Fusion/Stream/Monadic.hs:1449:1: warning: [-Wredundant-constraints]
    • Redundant constraint: Ord a
    • In the type signature for:
           enumFromTo_double :: (Monad m, Ord a, RealFrac a) =>
                                a -> a -> Stream m a
[ 8 of 21] Compiling Data.Vector.Fusion.Bundle.Monadic ( Data/Vector/Fusion/Bundle/Monadic.hs, dist/build/Data/Vector/Fusion/Bundle/Monadic.o )

Data/Vector/Fusion/Bundle/Monadic.hs:945:1: warning: [-Wredundant-constraints]
    • Redundant constraint: Ord a
    • In the type signature for:
           enumFromTo_double :: (Monad m, Ord a, RealFrac a) =>
                                a -> a -> Bundle m v a
[ 9 of 21] Compiling Data.Vector.Fusion.Bundle ( Data/Vector/Fusion/Bundle.hs, dist/build/Data/Vector/Fusion/Bundle.o )

Data/Vector/Fusion/Bundle.hs:114:1: warning: [-Wtabs]
    Tab character found here.
    Please use spaces instead.
[10 of 21] Compiling Data.Vector.Generic.Mutable ( Data/Vector/Generic/Mutable.hs, dist/build/Data/Vector/Generic/Mutable.o )
[11 of 21] Compiling Data.Vector.Generic.New ( Data/Vector/Generic/New.hs, dist/build/Data/Vector/Generic/New.o )
[12 of 21] Compiling Data.Vector.Generic ( Data/Vector/Generic.hs, dist/build/Data/Vector/Generic.o )

Data/Vector/Generic.hs:2080:1: warning: [-Wredundant-constraints]
    • Redundant constraint: Vector v a
    • In the type signature for:
           dataCast :: (Vector v a, Data a, Typeable v, Typeable t) =>
                       (forall d. Data d => c (t d)) -> Maybe (c (v a))
[13 of 21] Compiling Data.Vector.Primitive.Mutable ( Data/Vector/Primitive/Mutable.hs, dist/build/Data/Vector/Primitive/Mutable.o )
[14 of 21] Compiling Data.Vector.Primitive ( Data/Vector/Primitive.hs, dist/build/Data/Vector/Primitive.o )
[15 of 21] Compiling Data.Vector.Storable.Mutable ( Data/Vector/Storable/Mutable.hs, dist/build/Data/Vector/Storable/Mutable.o )

Data/Vector/Storable/Mutable.hs:72:1: warning: [-Wunused-imports]
    The import of ‘Foreign.Ptr’ is redundant
      except perhaps to import instances from ‘Foreign.Ptr’
    To import instances alone, use: import Foreign.Ptr()

Data/Vector/Storable/Mutable.hs:491:1: warning: [-Wredundant-constraints]
    • Redundant constraint: Storable a
    • In the type signature for:
           unsafeFromForeignPtr0 :: Storable a =>
                                    ForeignPtr a -> Int -> MVector s a

Data/Vector/Storable/Mutable.hs:501:1: warning: [-Wredundant-constraints]
    • Redundant constraint: Storable a
    • In the type signature for:
           unsafeToForeignPtr :: Storable a =>
                                 MVector s a -> (ForeignPtr a, Int, Int)

Data/Vector/Storable/Mutable.hs:511:1: warning: [-Wredundant-constraints]
    • Redundant constraint: Storable a
    • In the type signature for:
           unsafeToForeignPtr0 :: Storable a =>
                                  MVector s a -> (ForeignPtr a, Int)

Data/Vector/Storable/Mutable.hs:518:1: warning: [-Wredundant-constraints]
    • Redundant constraint: Storable a
    • In the type signature for:
           unsafeWith :: Storable a => IOVector a -> (Ptr a -> IO b) -> IO b
[16 of 21] Compiling Data.Vector.Storable ( Data/Vector/Storable.hs, dist/build/Data/Vector/Storable.o )

Data/Vector/Storable.hs:1408:1: warning: [-Wredundant-constraints]
    • Redundant constraint: Storable a
    • In the type signature for:
           unsafeFromForeignPtr0 :: Storable a =>
                                    ForeignPtr a -> Int -> Vector a

Data/Vector/Storable.hs:1417:1: warning: [-Wredundant-constraints]
    • Redundant constraint: Storable a
    • In the type signature for:
           unsafeToForeignPtr :: Storable a =>
                                 Vector a -> (ForeignPtr a, Int, Int)

Data/Vector/Storable.hs:1426:1: warning: [-Wredundant-constraints]
    • Redundant constraint: Storable a
    • In the type signature for:
           unsafeToForeignPtr0 :: Storable a =>
                                  Vector a -> (ForeignPtr a, Int)

Data/Vector/Storable.hs:1432:1: warning: [-Wredundant-constraints]
    • Redundant constraint: Storable a
    • In the type signature for:
           unsafeWith :: Storable a => Vector a -> (Ptr a -> IO b) -> IO b
[17 of 21] Compiling Data.Vector.Unboxed.Base ( Data/Vector/Unboxed/Base.hs, dist/build/Data/Vector/Unboxed/Base.o )

Data/Vector/Unboxed/Base.hs:362:10: warning: [-Wredundant-constraints]
    • Redundant constraint: RealFloat a
    • In the instance declaration for ‘M.MVector MVector (Complex a)’
[18 of 21] Compiling Data.Vector.Unboxed ( Data/Vector/Unboxed.hs, dist/build/Data/Vector/Unboxed.o )

internal/unbox-tuple-instances:133:1: warning: [-Wredundant-constraints]
    • Redundant constraints: (Unbox a, Unbox b)
    • In the type signature for:
           unzip :: (Unbox a, Unbox b) =>
                    Vector (a, b) -> (Vector a, Vector b)

internal/unbox-tuple-instances:307:1: warning: [-Wredundant-constraints]
    • Redundant constraints: (Unbox a, Unbox b, Unbox c)
    • In the type signature for:
           unzip3 :: (Unbox a, Unbox b, Unbox c) =>
                     Vector (a, b, c) -> (Vector a, Vector b, Vector c)

internal/unbox-tuple-instances:526:1: warning: [-Wredundant-constraints]
    • Redundant constraints: (Unbox a, Unbox b, Unbox c, Unbox d)
    • In the type signature for:
           unzip4 :: (Unbox a, Unbox b, Unbox c, Unbox d) =>
                     Vector (a, b, c, d) -> (Vector a, Vector b, Vector c, Vector d)

internal/unbox-tuple-instances:803:1: warning: [-Wredundant-constraints]
    • Redundant constraints: (Unbox a,
                              Unbox b,
                              Unbox c,
                              Unbox d,
                              Unbox e)
    • In the type signature for:
           unzip5 :: (Unbox a, Unbox b, Unbox c, Unbox d, Unbox e) =>
                     Vector (a, b, c, d, e)
                     -> (Vector a, Vector b, Vector c, Vector d, Vector e)

internal/unbox-tuple-instances:1121:1: warning: [-Wredundant-constraints]
    • Redundant constraints: (Unbox a,
                              Unbox b,
                              Unbox c,
                              Unbox d,
                              Unbox e,
                              Unbox f)
    • In the type signature for:
           unzip6 :: (Unbox a, Unbox b, Unbox c, Unbox d, Unbox e, Unbox f) =>
                     Vector (a, b, c, d, e, f)
                     -> (Vector a, Vector b, Vector c, Vector d, Vector e, Vector f)
[19 of 21] Compiling Data.Vector.Unboxed.Mutable ( Data/Vector/Unboxed/Mutable.hs, dist/build/Data/Vector/Unboxed/Mutable.o )

internal/unbox-tuple-instances:117:1: warning: [-Wredundant-constraints]
    • Redundant constraints: (Unbox a, Unbox b)
    • In the type signature for:
           unzip :: (Unbox a, Unbox b) =>
                    MVector s (a, b) -> (MVector s a, MVector s b)

internal/unbox-tuple-instances:282:1: warning: [-Wredundant-constraints]
    • Redundant constraints: (Unbox a, Unbox b, Unbox c)
    • In the type signature for:
           unzip3 :: (Unbox a, Unbox b, Unbox c) =>
                     MVector s (a, b, c) -> (MVector s a, MVector s b, MVector s c)

internal/unbox-tuple-instances:493:1: warning: [-Wredundant-constraints]
    • Redundant constraints: (Unbox a, Unbox b, Unbox c, Unbox d)
    • In the type signature for:
           unzip4 :: (Unbox a, Unbox b, Unbox c, Unbox d) =>
                     MVector s (a, b, c, d)
                     -> (MVector s a, MVector s b, MVector s c, MVector s d)

internal/unbox-tuple-instances:756:1: warning: [-Wredundant-constraints]
    • Redundant constraints: (Unbox a,
                              Unbox b,
                              Unbox c,
                              Unbox d,
                              Unbox e)
    • In the type signature for:
           unzip5 :: (Unbox a, Unbox b, Unbox c, Unbox d, Unbox e) =>
                     MVector s (a, b, c, d, e)
                     -> (MVector s a, MVector s b, MVector s c, MVector s d,
                         MVector s e)

internal/unbox-tuple-instances:1066:1: warning: [-Wredundant-constraints]
    • Redundant constraints: (Unbox a,
                              Unbox b,
                              Unbox c,
                              Unbox d,
                              Unbox e,
                              Unbox f)
    • In the type signature for:
           unzip6 :: (Unbox a, Unbox b, Unbox c, Unbox d, Unbox e, Unbox f) =>
                     MVector s (a, b, c, d, e, f)
                     -> (MVector s a, MVector s b, MVector s c, MVector s d,
                         MVector s e, MVector s f)
[20 of 21] Compiling Data.Vector.Mutable ( Data/Vector/Mutable.hs, dist/build/Data/Vector/Mutable.o )
[21 of 21] Compiling Data.Vector      ( Data/Vector.hs, dist/build/Data/Vector.o )

Data/Vector.hs:223:14: error:
    Ambiguous occurrence ‘fromList’
    It could refer to either ‘Exts.fromList’,
                             imported from ‘Data.Primitive.Array’ at Data/Vector.hs:164:1-37
                             (and originally defined in ‘GHC.Exts’)
                          or ‘Data.Vector.fromList’, defined at Data/Vector.hs:1528:1

Data/Vector.hs:224:15: error:
    Ambiguous occurrence ‘fromListN’
    It could refer to either ‘Exts.fromListN’,
                             imported from ‘Data.Primitive.Array’ at Data/Vector.hs:164:1-37
                             (and originally defined in ‘GHC.Exts’)
                          or ‘Data.Vector.fromListN’, defined at Data/Vector.hs:1537:1

Data/Vector.hs:344:19: error:
    Ambiguous occurrence ‘fromList’
    It could refer to either ‘Exts.fromList’,
                             imported from ‘Data.Primitive.Array’ at Data/Vector.hs:164:1-37
                             (and originally defined in ‘GHC.Exts’)
                          or ‘Data.Vector.fromList’, defined at Data/Vector.hs:1528:1
cabal: Leaving directory '/tmp/cabal-tmp-26528/vector-0.11.0.0'
cabal: Error: some packages failed to install:
vector-0.11.0.0 failed during the building phase. The exception was:
ExitFailure 1

```

[This commit](https://github.com/haskell/primitive/commit/7c7afa414356b7c297d428355350845ff239a40c) in `primitive` appears to be responsible.

**Comment by geraldus** [comment]
Observing same issue

**Comment by phadej** [comment]
queued build on matrix (the 3rd). Let's see.

**Comment by hvr** [comment]
I saw this confirmed on matrix3 and did the edits:

 - https://hackage.haskell.org/package/vector-0.11.0.0/revisions/
 - https://hackage.haskell.org/package/vector-0.10.12.3/revisions/

**Comment by Gabriel439** [comment]
The latter `fromList` error still persists.  I got this build failure today:

```
Configuring vector-0.11.0.0...
Building vector-0.11.0.0...
Preprocessing library vector-0.11.0.0...
[ 1 of 21] Compiling Data.Vector.Storable.Internal ( Data/Vector/Storable/Internal.hs, dist/build/Data/Vector/Storable/Internal.o )
[ 2 of 21] Compiling Data.Vector.Generic.Mutable.Base ( Data/Vector/Generic/Mutable/Base.hs, dist/build/Data/Vector/Generic/Mutable/Base.o )
[ 3 of 21] Compiling Data.Vector.Generic.Base ( Data/Vector/Generic/Base.hs, dist/build/Data/Vector/Generic/Base.o )
[ 4 of 21] Compiling Data.Vector.Fusion.Util ( Data/Vector/Fusion/Util.hs, dist/build/Data/Vector/Fusion/Util.o )
[ 5 of 21] Compiling Data.Vector.Fusion.Bundle.Size ( Data/Vector/Fusion/Bundle/Size.hs, dist/build/Data/Vector/Fusion/Bundle/Size.o )
[ 6 of 21] Compiling Data.Vector.Internal.Check ( Data/Vector/Internal/Check.hs, dist/build/Data/Vector/Internal/Check.o )
[ 7 of 21] Compiling Data.Vector.Fusion.Stream.Monadic ( Data/Vector/Fusion/Stream/Monadic.hs, dist/build/Data/Vector/Fusion/Stream/Monadic.o )
[ 8 of 21] Compiling Data.Vector.Fusion.Bundle.Monadic ( Data/Vector/Fusion/Bundle/Monadic.hs, dist/build/Data/Vector/Fusion/Bundle/Monadic.o )
[ 9 of 21] Compiling Data.Vector.Fusion.Bundle ( Data/Vector/Fusion/Bundle.hs, dist/build/Data/Vector/Fusion/Bundle.o )
[10 of 21] Compiling Data.Vector.Generic.Mutable ( Data/Vector/Generic/Mutable.hs, dist/build/Data/Vector/Generic/Mutable.o )
[11 of 21] Compiling Data.Vector.Generic.New ( Data/Vector/Generic/New.hs, dist/build/Data/Vector/Generic/New.o )
[12 of 21] Compiling Data.Vector.Generic ( Data/Vector/Generic.hs, dist/build/Data/Vector/Generic.o )
[13 of 21] Compiling Data.Vector.Primitive.Mutable ( Data/Vector/Primitive/Mutable.hs, dist/build/Data/Vector/Primitive/Mutable.o )
[14 of 21] Compiling Data.Vector.Primitive ( Data/Vector/Primitive.hs, dist/build/Data/Vector/Primitive.o )
[15 of 21] Compiling Data.Vector.Storable.Mutable ( Data/Vector/Storable/Mutable.hs, dist/build/Data/Vector/Storable/Mutable.o )
[16 of 21] Compiling Data.Vector.Storable ( Data/Vector/Storable.hs, dist/build/Data/Vector/Storable.o )
[17 of 21] Compiling Data.Vector.Unboxed.Base ( Data/Vector/Unboxed/Base.hs, dist/build/Data/Vector/Unboxed/Base.o )
[18 of 21] Compiling Data.Vector.Unboxed ( Data/Vector/Unboxed.hs, dist/build/Data/Vector/Unboxed.o )
[19 of 21] Compiling Data.Vector.Unboxed.Mutable ( Data/Vector/Unboxed/Mutable.hs, dist/build/Data/Vector/Unboxed/Mutable.o )
[20 of 21] Compiling Data.Vector.Mutable ( Data/Vector/Mutable.hs, dist/build/Data/Vector/Mutable.o )
[21 of 21] Compiling Data.Vector      ( Data/Vector.hs, dist/build/Data/Vector.o )

Data/Vector.hs:223:14:
    Ambiguous occurrence ‘fromList’
    It could refer to either ‘Data.Vector.fromList’,
                             defined at Data/Vector.hs:1528:1
                          or ‘Exts.fromList’,
                             imported from ‘Data.Primitive.Array’ at Data/Vector.hs:164:1-37
                             (and originally defined in ‘GHC.Exts’)

Data/Vector.hs:224:15:
    Ambiguous occurrence ‘fromListN’
    It could refer to either ‘Data.Vector.fromListN’,
                             defined at Data/Vector.hs:1537:1
                          or ‘Exts.fromListN’,
                             imported from ‘Data.Primitive.Array’ at Data/Vector.hs:164:1-37
                             (and originally defined in ‘GHC.Exts’)

Data/Vector.hs:344:19:
    Ambiguous occurrence ‘fromList’
    It could refer to either ‘Data.Vector.fromList’,
                             defined at Data/Vector.hs:1528:1
                          or ‘Exts.fromList’,
                             imported from ‘Data.Primitive.Array’ at Data/Vector.hs:164:1-37
                             (and originally defined in ‘GHC.Exts’)
Failed to install vector-0.11.0.0
```

**Comment by hvr** [comment]
@Gabriel439 did you `cabal update`?

**Comment by Gabriel439** [comment]
@hvr Yes.  This is using your Travis script generator and I believe this is where it runs `cabal update`:

https://github.com/Gabriel439/Haskell-Turtle-Library/blob/master/.travis.yml#L41

Here's the corresponding failing job:

https://travis-ci.org/Gabriel439/Haskell-Turtle-Library/builds/177555674

**Comment by Gabriel439** [comment]
Never mind.  The problem has disappeared in the latest run of the CI job:

https://travis-ci.org/Gabriel439/Haskell-Turtle-Library/builds/177555674

**Comment by RyanGlScott** [comment]
It looks like this issue is fixed in `vector` HEAD, so I suppose a fix would be to release a new version of `vector`.

**Comment by dolio** [comment]
Sorry. Things have been a bit hectic. I'll try to get a release together soon.

**Comment by bgamari** [comment]
It would be great to have a release. @dolio, can I be of help with this?

**Comment by RyanGlScott** [comment]
Since `vector-0.12.0.0` has been released with support for `primitive-0.6.2.0`, I'd say this issue has been resolved.

**Comment by juhp** [comment]
Thanks, note that Stackage Nightly won't be able to update for a while yet (fpco/stackage#2194).
